lib/eventemitter2.js

```diff
@@ -782,7 +782,8 @@
     module.exports = EventEmitter;
   }
   else {
-    // Browser global.
-    window.EventEmitter2 = EventEmitter;
+    // global for any kind of environment.
+    var _global= new Function('','return this')();
```

Comment:
Are you certain this will work in all JS contexts? What do you think about using the [global](https://www.npmjs.com/package/global) NPM module for this?

Reply:
This is a well-known (https://stackoverflow.com/questions/3277182/how-to-get-the-global-object-in-javascript) approach over the past 10 years that I have often used, starting from IE7 to the present. As we known, “this” always refers to the global object in non-strict mode. It works with old IE, works with node, modern browsers and with workers of any kind (browser&node). I have never heard of problems with this pattern. But if you think we should add a “global” package to the dependencies for this, I can do it.

Reply:
I'm not against using it, just wasn't sure how universal it was. You mentioned strict mode, does that affect whether this works in an ESM module context? I could see people importing a build of eventemitter2 in ESM modules as those get more popular. Not sure if `global` accounts for it though. 🤔

Reply:
By default, the code runs in non-strict mode in all environments (except ESM). You need to specify “use strict” to enable it only for the local scope (script | function | module), but you can't do this for the whole global scope, the global scope always works in non-strict mode due to maintaining backward compatibility with the old code base. So this trick will work as long as T39 decides to break backward compatibility and by default enables strict mode for the global scope. But since no one really needs this, they made the “use strict” directive and we still using it.
A function constructor always runs in the global scope, so switching a user-defined function to strict mode or using ES modules with strict mode enabled by default does not affect the global scope, which always in non-strict mode. So, this approach must works in all environments, despite the strict mode enabled inside modules, scripts or functions.
```
    (function(){
        "use strict"; //enable strict mode
        console.log(this); //this===undefined we are in strict mode inside this function
        console.log((function(){return this})()); //this===undefined we are still in strict mode
        console.log(new Function('return this')()) //but this function runs in global non-strict mode, so it returns window | global
    })();
```

Reply:
Wonderful, thank you for the explanation!

Reply:
Great! Thank you for your cooperation :)